[WIP] Pass SELinux Mount label down to graphdriver

rhatdan opened the pull request:
Currently when we do a commmit, we are mounting the container without using
the mountlabel. In certain situations we can leak mount points where the
image is already mounted with a label. If you then attempt to commit the
image, the kernel will attempt to mount the image without a label. The
kernel will reject this mount since SELinux does not allow the same image
to be mounted with different labels.

Passing down the label to the diff drivers, fixes this issue.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

coolljt0725 commented:
@rhatdan  It's `WIP`, are you still work on this?

rhatdan commented:
@coolljt0725 Yes, although I was away at a conference for a couple of weeks.  Hope to get back to it soon.

coolljt0725 commented:
@rhatdan  Thanks  ：）

rhatdan commented:
@rhvgoyal PTAL

rhatdan commented:
@coolljt0725 Attempting to figure out how to handle mountlabel through the layerstore.

dmcgowan commented:
> Attempting to figure out how to handle mountlabel through the layerstore.

There aren't too many options here, the mounted layer objects must be able to be recreated based on the on-disk metadata. Expanding the exported interface is not an option but storing a non-empty mount label along with the other metadata could be possible. Once #36504 is merged this will no longer require an interface change.

In the long term, this would probably just end up as labels inside the containerd metadata store and the Docker client can interpret them and apply the labels however it wants.

rhatdan commented:
Well the running container labels and the committed container labels need to match.  If they don't or we have a leaked mount point from a running container then the diff mount will fail.

rhatdan committed:
Pass SELinux Mount label down to graphdriver

Currently when we do a commmit, we are mounting the container without using
the mountlabel. In certain situations we can leak mount points where the
image is already mounted with a label. If you then attempt to commit the
image, the kernel will attempt to mount the image without a label. The
kernel will reject this mount since SELinux does not allow the same image
to be mounted with different labels.

Passing down the label to the diff drivers, fixes this issue.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

olljanat commented:
Derek add label: rebuild/*
